staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -231,14 +231,19 @@ func (cf ConditionFunc) WithContext() ConditionWithContextFunc {
 // Note the caller must *always* call the CancelFunc, otherwise resources may be leaked.
 func ContextForChannel(parentCh <-chan struct{}) (context.Context, context.CancelFunc) {
 	ctx, cancel := context.WithCancel(context.Background())
-
-	go func() {
-		select {
-		case <-parentCh:
-			cancel()
-		case <-ctx.Done():
-		}
-	}()
+	select {
```

Comment:
@smarterclayton this PR was superseded by https://github.com/kubernetes/kubernetes/pull/115140, right?

Reply:
Correct, let me address your review comments first and I'll update.